README.md

```diff
@@ -2381,6 +2383,46 @@ Notice that the [`Browser`](#browser) is an immutable object, i.e. this
 method actually returns a *new* [`Browser`](#browser) instance with the
 given setting applied.
 
+#### withHeader()
+
+The `withHeader(string $header, string $value): Browser` method can be used to
+add a request header for all following requests.
+
+```php
+$browser = $browser->withHeader('User-Agent', 'ACME');
+
+$browser->get($url)->then(function (Psr\Http\Message\ResponseInterface $response) {
+    //will contain the header User-Agent with the value ACME
+    var_dump($response->getHeaders());
```

Comment:
I think it can be confusing that we show two types of headers in this example (request and response).
```suggestion
    var_dump($response);
```